src/Mantasflowers.Services/Services/User/UserService.cs

```diff
@@ -83,6 +83,17 @@ public async Task<string> GetUserUidByGuidAsync(Guid id)
             return user?.Uid;
         }
 
+        public async Task<Guid> GetUserGuidByUidAsync(string uid)
+        {
+            var user = await _unitOfWork.UserRepository.GetUserByUidAsync(uid);
+            if (user == null)
+            {
+                throw new FirebaseUidNotFoundException($"No user could be found for {nameof(uid)}");
+            }
+
+            return user.Id;
+        }
+
```

Comment:
You don't need this method. Just call `GetUserByUidAsync()` and take `Id` from the returned value.

Reply:
It's more for convenience than the sake of the argument. Abstracting away details from a PaymentController.

Reply:
But `GetUserByUidAsync()` doesn't return an entity. It returns a mapper response which is already abstraction enough.

Reply:
I guess, it is to show, that only id is required and nothing else.

Reply:
By that logic you can also have methods like `GetUserFirstNameByUid`, `GetUserLastNameByUid`, `GetUserAgeByUid` ? 🤔

Reply:
Which I don't need 🍡 That whole part with linking user to order was a pain to realize without cycling dependencies on services and keeping whole operation as a unit.